### ISSUE TITLE
integration with OperationResultTools library

### ISSUE THREAD

**Issue by N1K0232** [marcominerva/StorageProviders#18]
integration with OperationResultTools library to manage errors when saving, reading and deleting

example:
public async Task<Result> SaveAsync(string path, Stream stream)
{
        // ...
}

**Comment by marcominerva** [comment]
Can you explain better your scenario?

**Comment by N1K0232** [comment]
Yeah sure. I apologize both for being late and for not giving the explanation. By mistake I didn't change the return type

I was meaning a scenario just like this:
public async Task<Result> SaveAsync(string path, Stream stream, bool overwrite = false, CancellationToken cancellationToken = default)
    {
       if(string.IsNullOrWhiteSpace(path))
       {
           return Result.Fail(FailureReason.ClientError, "the path is required");
       }

       if(stream is null)
       {
          return Result.Fail(FailureReason.ClientError, "the stream is required");
       }

        try
       {
           var blobClient = await GetBlobClientAsync(path, true, cancellationToken).ConfigureAwait(false);

            if (!overwrite)
            {
                var blobExists = await blobClient.ExistsAsync(cancellationToken).ConfigureAwait(false);
                if (blobExists)
                {
                   //instead of throwing the exception
                    return Result.Fail(FailureReason.GenericError, $"The file {path} already exists");
                    //throw new IOException($"The file {path} already exists.");
                }
            }
    
            if (stream.CanSeek)
            {
                stream.Position = 0;
            }
    
            await blobClient.UploadAsync(stream, new BlobHttpHeaders { ContentType = MimeUtility.GetMimeMapping(path) }, cancellationToken: cancellationToken).ConfigureAwait(false);

           return Result.Ok();
       }
       catch(Exception ex)
       {
             //I used Exception common type just for the example
             return Result.Fail(FailureReason.GenericError, ex); 
       }
    }

I hope this example can explain better my scenario
